src/rocks_parameters.cpp

```diff
@@ -151,5 +154,56 @@ namespace mongo {
         _engine->getBlockCache()->SetCapacity(newSizeInBytes);
 
         return Status::OK();
+    }    
+    
+    RocksOptionsParameter::RocksOptionsParameter(RocksEngine* engine)
+        : ServerParameter(ServerParameterSet::getGlobal(), "rocksdbOptions", false,
+                          true),
+          _engine(engine) {}
+
+    void RocksOptionsParameter::append(OperationContext* txn, BSONObjBuilder& b,
+                                         const std::string& name) {
+        std::string columnOptions;
+        std::string dbOptions;
+        std::string fullOptionsStr;
+        rocksdb::Options fullOptions = _engine->getDB()->GetOptions();
+        rocksdb::Status s = GetStringFromColumnFamilyOptions(&columnOptions, fullOptions);
+        if (!s.ok()) { // If we failed, append the error for the user to see.
+            b.append(name, s.ToString()); 
+            return;
+        }
+        
+        fullOptionsStr.append(columnOptions);
+        
+        s = GetStringFromDBOptions(&dbOptions, fullOptions);
+        if (!s.ok()) { // If we failed, append the error for the user to see.
+            b.append(name, s.ToString()); 
+            return;
+        }
+        
+        fullOptionsStr.append(dbOptions);
+
+        b.append(name, fullOptionsStr);
+    }
+
+    Status RocksOptionsParameter::set(const BSONElement& newValueElement) {        
+        return setFromString(newValueElement.String());
```

Comment:
OK, in that case it'd be helpful to add a comment mentioning that above L190. Other than that, the PR looks great!